src/shared_modules/utils/socketWrapper.hpp

```diff
@@ -110,6 +110,163 @@ class TcpAddress final : public SockAddress<TcpAddress>
+                            uint32_t sizeHeader = 0)
+    {
+        if (PACKET_SIZE + HEADER_SIZE + sizeHeader + sizeBody > BUFFER_SIZE)
+        {
+            buffer.resize(PACKET_SIZE + HEADER_SIZE + sizeHeader + sizeBody + 1);
+        }
+
+        // Write packet size to the buffer.
+        uint32_t* uip;
+        uip = (uint32_t*)buffer.data();
+        *uip = sizeBody + HEADER_SIZE + sizeHeader;
+
+        // Write header size to the buffer.
+        uip = (uint32_t*)(buffer.data() + PACKET_SIZE);
+        *uip = sizeHeader;
+
+        if (sizeHeader > 0)
+        {
+            std::copy(dataHeader, dataHeader + sizeHeader, std::begin(buffer) + PACKET_SIZE + HEADER_SIZE);
+        }
+
+        std::copy(dataBody, dataBody + sizeBody, std::begin(buffer) + PACKET_SIZE + HEADER_SIZE + sizeHeader);
+
+        bufferSize = PACKET_SIZE + HEADER_SIZE + sizeHeader + sizeBody;
+    }
+
+    /**
+     * @brief Get the size of the header according to this protocol.
+     *
+     * @param buffer Buffer to obtain the header size from.
+     * @return auto Header size.
+     */
+    auto static getHeaderSize(const std::vector<char>& buffer)
+    {
+        return *reinterpret_cast<const uint32_t*>(buffer.data());
+    }
+
+    /**
+     * @brief Get the data offset according to this protocol.
+     *
+     * @param headerSize The size of the header.
+     * @return auto Data offset.
+     */
+    auto static getDataOffset(uint32_t headerSize)
+    {
+        return HEADER_SIZE + headerSize;
+    }
+
+    /**
+     * @brief Get the header offset according to this protocol.
+     *
+     * @return auto Header offset.
+     */
+    auto static getHeaderOffset()
+    {
+        return HEADER_SIZE;
+    }
+};
+
+/**
+ * @brief This class is used for the following format:
+ *          - 4 bytes: header size. Mandatory size of the data to send.
+ *          - M bytes: body. Mandatory data to send.
+ */
+class sizeHeaderProtocol final
+{
+public:
+    /**
+     * @brief Create the buffer to send according to this protocol.
+     *
+     * @param buffer        Output buffer to write.
+     * @param bufferSize    Size of the output buffer.
+     * @param dataBody      Data to send.
+     * @param sizeBody      Size of the data to send.
+     * @param dataHeader    Optional header to send.
+     * @param sizeHeader    Size of the optional header.
+     */
+    void static buildBuffer(std::vector<char>& buffer,
+                            uint32_t& bufferSize,
+                            const char* dataBody,
+                            uint32_t sizeBody,
+                            const char* dataHeader = nullptr,
+                            uint32_t sizeHeader = 0)
+    {
+        if (PACKET_SIZE + sizeBody > BUFFER_SIZE)
+        {
+            buffer.resize(PACKET_SIZE + sizeBody + 1);
+        }
+
+        // Write packet size to the buffer.
+        uint32_t* uip;
+        uip = (uint32_t*)buffer.data();
+        *uip = sizeBody;
```

Comment:
For transmitting data between different machines with probably different endianness, "network byte order" is preferred.
https://www.ibm.com/docs/en/zvm/7.2?topic=domains-network-byte-order-host-byte-order
For modifying host to network byte order and vice-versa, the following library function could be used:
https://man7.org/linux/man-pages/man3/htons.3.html